File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        vector<double> ans;
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()){
            double row{0};
            double qlen = q.size();
            for (int i = 0; i < qlen; ++i){
                TreeNode* cur = q.front();
                q.pop();
                row += cur->val;
                if (cur->left) q.push(cur->left);
                if (cur->right) q.push(cur->right);
            }
            ans.push_back(row/qlen);
        }
        return ans;
    }
};

File: 0637-average-of-levels-in-binary-tree/0637-average-of-levels-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        q, info = [root], []
        while len(q):
            qlen, row = len(q), 0
            for i in range(qlen):
                cur = q.pop(0)
                row += cur.val
                if cur.left: q.append(cur.left)
                if cur.right: q.append(cur.right)
            info.append(row/qlen)
        return info